Fix, and update contributing guide

marcalath opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
Fixed the broken links in the documentation, following the discussion in https://github.com/moby/moby/pull/44826 

**- How I did it**
Added the missing documentation mentioned, fixed the links and added an updated image.

**- How to verify it**
Click the links.

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
Fix broken contributing documentation links

fixes  #45024

**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://user-images.githubusercontent.com/125604796/219528011-9eeb0a00-4c2f-4853-a768-9bff1982703b.png)

marcalath committed:
Add the advanced-contributing page

Following the discussion on
https://github.com/moby/moby/pull/44826
and its summary in CONTRIBUTING.md,
the advanced-contributing page was added.
Based on
https://github.com/docker/docs/blob/7cfe2e9da240cd9d2dc475593af1a4840a518614/opensource/workflow/advanced-contributing.md
and adjusted for moby.

Signed-off-by: Snir Ilani <marcalath@gmail.com>

marcalath committed:
Fix an old link in the contribution docs

Link updated from
https://docs.docker.com/engine/installation/
to
https://docs.docker.com/engine/install/
and inline HTML formatting was replaced by
md link formatting.

Signed-off-by: Snir Ilani <marcalath@gmail.com>

marcalath committed:
Add the design PR process image

Referenced in
docs/contributing/advanced-contributing.md

Signed-off-by: Snir Ilani <marcalath@gmail.com>

marcalath committed:
Fix broken links on CONTRIBUTING.md

Broken docs.docker.com links were
replaced with updated ones.
Community Slack link was updated.
Doc-style redirects to Docker's grammer
reference - which presents a similar notion.

Signed-off-by: Snir Ilani <marcalath@gmail.com>

marcalath committed:
Align contribution docs with header format

Signed-off-by: Snir Ilani <marcalath@gmail.com>

neersighted commented on CONTRIBUTING.md:
```diff
@@ -101,7 +101,7 @@ the contributors guide.
       <p>
         Register for the Docker Community Slack at
-	<a href="https://dockr.ly/slack" target="_blank">https://dockr.ly/slack</a>.
+	<a href="https://dockr.ly/slack" target="_blank">https://dockr.ly/comm-slack</a>.
```
The actual `href` was not updated.

marcalath replied:
nice catch :)

neersighted commented on CONTRIBUTING.md:
```diff
@@ -4,7 +4,7 @@ Want to hack on the Moby Project? Awesome! We have a contributor's guide that ex
 process](docs/contributing/). 
 
-[![Contributors guide](docs/static_files/contributors.png)](https://docs.docker.com/opensource/project/who-written-for/)
+[![Contributors guide](docs/static_files/contributors.png)](https://docs.docker.com/contribute/overview/)
```
This is likely still the wrong link; this guide is for the docs only. We should probably link to `who-written-for.md` or something else instead (or kill the image).

marcalath replied:
OK, sure, we can do that and just update the image

neersighted commented on docs/contributing/advanced-contributing.md:
```diff
@@ -0,0 +1,141 @@
+
+3. Discuss with the community:
+
+    Check out our [community page](https://mobyproject.org/community/), where you'll find links to our [Twitter account](https://twitter.com/moby), [Slack channel](https://dockr.ly/comm-slack) and [forum](https://forums.mobyproject.org/).
```
@thaJeztah do we still want to plug the forum or twitter?

neersighted commented on docs/contributing/advanced-contributing.md:
```diff
@@ -0,0 +1,141 @@
+
+    PR's title's format:
+
+    **Refactor:** _short title_
```
Our current convention is `area/of/code: short title` or `Short title` if it's further-reaching than one section of the codebase.

neersighted reviewed (changes requested):
Thanks for taking this one! Some initial feedback; and also I'm not sure we currently want to push people to do the "PR a markdown proposal" workflow as we have not actually done that for some time.

Still, it's a lot of work to update this, thanks so much for contributing and working on resolving this sharp edge :smile:

marcalath committed:
fix contributing docs

Following the review process in #45026:

>> CONTRIBUTING.md <<
- Slack link was fixed in href too
- image was brushed and now it does not
include an in-image-link
- image now links to who-written-for.md

>> advanced-contributing.md <<
- PR title format updated

Signed-off-by: Snir Ilani <marcalath@gmail.com>

marcalath commented:
Hey @neersighted , sure, I'm happy to help :)
Also, could you elaborate on "PR a markdown proposal"?
Would you prefer that we have a separate process for doc PRs?

neersighted commented:
In `advanced-contributing.md` there is a workflow described that involves first creating a PR with a write-up of a proposed design, and then implementing after that PR is accepted. I don't know that we currently follow that workflow; if we want to return to it, it makes sense, but if it's not currently accurate we should update the content.

marcalath commented:
Oh, OK :) So I'm guessing we'll wait for more maintainers to comment on this one - @thaJeztah maybe?

neersighted commented:
Precisely, no action needed from you until we figure out what workflow we want to document :laughing:

marcalath committed:
Fix numbering in set-up-ide.md

Signed-off-by: Snir Ilani <marcalath@gmail.com>

marcalath committed:
Fix the contribution guide flow

The contribution guide now has a unified flow:

- who-written-for.md was more appropriately
renamed getting-started.md
and links to the file were updated.

- The links in the different contribution docs
were updated, and form a unified flow.
see docs/contributing/README.md for the steps.

- doc files were updated with links to the following
step in the guide to form the unified flow.

Signed-off-by: Snir Ilani <marcalath@gmail.com>

robmry commented:
These changes look great ... I've got some mostly-separate minor tweaks lined up, but I'll hold off until this PR is merged.

(But, I'm new here! So can't answer the questions that have been raised.)
